astropy/utils/state.py

```diff
@@ -3,10 +3,63 @@
 :ref:`config-developer` for more details.
 """
 
+import copyreg
+from types import MappingProxyType
 
 __all__ = ['ScienceState']
 
 
+def _reduce_mappingproxy(mpt: MappingProxyType):
+    """
+    Allow MappingProxyType to be deep copied.
+
+    Returns
+    -------
+    `dict`
+        The type
+    Tuple[Tuple[Tuple[Any, Any]]]
+        A 1-element tuple containing the tuple of the itemsview of
+        the MappingProxyType mappable.
+
+    """
+    return dict, (tuple(mpt.items()), )
+
+
+# register MappingProxyType copy method
+copyreg.pickle(MappingProxyType, _reduce_mappingproxy)
```

Comment:
Is this truly needed? It doesn't seem a good idea for astropy to be modifying how `MappingProxyType` is pickled - which I think would affect all packages that are being run. If we really need it, I would subclass it (I guess inside `galactocentric` - right now, this is too much action at a distance!!)

Reply:
Unfortunately, `MappingProxyType` cannot be subclassed. I have implemented a fix by subclassing `MappingView` with a `MappingProxy` object. Now the `deepcopy` works without any AaaD.

I chose to do it within `state` since my next PR, which will make all `ScienceStates` have `_state` for improved metadata support. Is that alright?

Reply:
Ideally, just do it in `galactocentric` - sorry to be making a hassle, but I think it is really best to separate steps...

Reply:
Fair enough.